operator/v1/types_csi_cluster_driver.go

```diff
@@ -118,25 +121,116 @@ const (
 type CSIDriverConfigSpec struct {
 	// driverType indicates type of CSI driver for which the
 	// driverConfig is being applied to.
-	//
-	// Valid values are:
-	//
-	// * vSphere
-	//
-	// Allows configuration of vsphere CSI driver topology.
-	//
-	// ---
+	// Valid values are: AWS, Azure, GCP, vSphere
```

Comment:
I don't know. I will look into this.

Reply:
If DriverType is "" then the contents of CSIDriverConfigSpec are ignored. It's a no-op.

Reply:
I think the way it's worded is correct:
```
	// Valid values are: AWS, Azure, GCP, vSphere
	// Consumers should treat unknown values as a NO-OP.
```
If you say `AWS, Azure, GCP, vSphere and omitted` instead then it sounds like the string "omitted" is also valid.

Reply:
We use `omitted` frequently through our API documentation to mean that the value isn't present. I would prefer we are consistent with our other optioanl Enums and keep the `and omitted` and then explain`, `When omitted, this means...` verbiage.

The Consumers note you've pointed out seems like it's aimed at developers, rather than end users? Right? 
If it's a developer note, prefix it with a `+` to remove it from the generated customer docs

Reply:
> We use omitted frequently through our API documentation to mean that the value isn't present.

Ok, then I have no objection.

> The Consumers note you've pointed out seems like it's aimed at developers, rather than end users? Right?
If it's a developer note, prefix it with a + to remove it from the generated customer docs

Yes, and with the validator, I can't set an unknown or invalid value as a user anyway.

So based on your suggestions, how about this?
```
	// Valid values are: AWS, Azure, GCP, vSphere and omitted.
	// When this value is omitted, the CSI driver config spec is not used.
	// +Consumers should treat unknown values as a NO-OP.
```

Reply:
Yeah I think that's good, but what is used when it is omitted, some defaults?